trulens_eval/tests/e2e/test_providers.py

```diff
@@ -784,5 +786,43 @@ def test_hugs_calibration(self):
             print(f"{h}: {total_tests}/{total_tests} tests passed.")
 
 
+    @optional_test
+    def test_langchain_feedback(self):
+        """
+        Check that LangChain feedback functions produce values within the expected range
+        and adhere to the expected format.
+        """
+        from trulens_eval.feedback.provider.langchain import LangChain
+        lc = LangChain()
+
+        tests = get_langchain_tests(lc)
+
+        failed_tests = lambda : len(failed_subtests)
+        total_tests = 0
+        failed_subtests = []
+
+        for imp, args, expected in tests:
+            subtest_name = f"{imp.__name__}-{args}"
+            actual = imp(**args)
+            with self.subTest(subtest_name):
+                total_tests += 1
+                try:
+                    self.assertAlmostEqual(actual, expected, delta=0.2)
+                except AssertionError:
+                    failed_subtests.append((subtest_name, actual, expected))
+
+        if failed_tests() > 0:
```

Comment:
I'm not sure if you need to do this yourself as I think unittest will give you a summary of failures, even with subtests.